Services/Administration/OJS.Services.Administration.Data/Implementations/ProblemGroupsDataService.cs

```diff
@@ -38,5 +41,9 @@ public IQueryable<ProblemGroup> GetAllByContestId(int contestId)
         public bool IsFromContestByIdAndContest(int id, int contestId) =>
             this.GetByIdQuery(id)
                 .Any(pg => pg.ContestId == contestId);
+        protected override Expression<Func<ProblemGroup, bool>> GetUserFilter(UserInfoModel user)
+            => problemGroup => user.IsAdmin ||
+                               problemGroup.Contest!.Category!.LecturersInContestCategories.Any(cc => cc.LecturerId == user.Id) ||
+                               problemGroup.Contest.LecturersInContests.Any(l => l.LecturerId == user.Id);
```

Comment:
See if you can utilize the `ILecturerContestPrivilegesBusinessService`

Reply:
I don't think so because the user permissions are taken inside the DataService